Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -1564,6 +1524,60 @@ public override void Push()
       SendThisVC("(push 1)");
       DeclCollector.Push();
     }
+    
+    
+  
+  
+    public class ProverProblemCollector
+    {
+      public ConcurrentQueue<string> Warnings = new();
+      private readonly ConcurrentStack<string> errors = new();
```

Comment:
I'm not sure I understand why this one's a stack instead of a queue. Just calling that out in case it's a mistake. If it's intentional, I'm otherwise happy with this PR.